AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m

```diff
@@ -67,23 +67,29 @@ - (BOOL)saveLog:(id<MSLog>)log withGroupId:(NSString *)groupId flags:(MSFlags)fl
                                    targetKey ? [NSString stringWithFormat:@"'%@'", targetKey] : @"NULL", (unsigned int)persistenceFlags];
   }
   return [self executeQueryUsingBlock:^int(void *db) {
+           // Selecting logs with equal or lower priority and ordering by priority then age.
            int result = [MSDBStorage executeNonSelectionQuery:addLogQuery inOpenedDatabase:db];
 
            // If the database is full, delete logs until there is room to add the log.
            long countOfLogsDeleted = 0;
-           while (result == SQLITE_FULL) {
-             result = [MSLogDBStorage deleteOldestLogsWithCount:1 inOpenedDatabase:db];
+           NSArray<NSNumber *> *ids = [self logsWithPriorityLowerThan:persistenceFlags inOpenedDatabase:db];
```

Comment:
Does this return logs ordered by age? If so, I think the method name should reflect that (for readability)

Reply:
And if not, I believe it should

Reply:
Remember that if we are getting SQLITE_FULL, we have already filled the database, which has a default max size of 10mb. So that means that in the default case, we are loading around 10mb of logs every time a single log is added (and this code path is executed). Further, we don't actually need all 10mb, we probably only need 1 or 2 logs.

This method seems to load all logs instead of the old way which loads them in chunks of 1. I propose we load them in chunks of, say, 5 logs at a time? That way we aren't loading all the logs in storage for every new log that comes through. (I think even just 1 log at a time would be better).

Does anyone else have any thoughts on that?

Reply:
It's also not just lower it can be equal priority :)

Reply:
Just few lines of code. I removed the method.